scripts/autolabel.lean

```diff
@@ -0,0 +1,286 @@
+
+-/
+
+open Lean System
+
+namespace AutoLabel
+
+/--
+A `Label` consists of the
+* The `label` field is the actual GitHub label name.
+* The `dirs` field is the array of all "root paths" such that a modification in a file contained
+  in one of these paths should be labelled with `label`.
+* The `exclusions` field is the array of all "root paths" that are excluded, among the
+  ones that start with the ones in `dirs`.
+  Any modifications to a file in an excluded path is ignored for the purposes of labelling.
+-/
+structure Label where
+  /-- The label name as it appears on GitHub -/
+  label : String
+  /-- Array of paths which fall under this label. e.g. `"Mathlib" / "Algebra"`.
+
+  For a label of the form `t-set-theory` this defaults to `#["Mathlib" / "SetTheory"]`. -/
+  dirs : Array FilePath := if label.startsWith "t-" then
+      #["Mathlib" / ("".intercalate (label.splitOn "-" |>.drop 1 |>.map .capitalize))]
+    else #[]
+  /-- Array of paths which should be excluded.
+  Any modifications to a file in an excluded path are ignored for the purposes of labelling. -/
+  exclusions : Array FilePath := #[]
+  deriving BEq, Hashable
+
+/--
+Mathlib labels and their corresponding folders. Add new labels and folders here!
+-/
+def mathlibLabels : Array Label := #[
+  { label := "t-algebra",
+    dirs := #[
+      "Mathlib" / "Algebra",
+      "Mathlib" / "FieldTheory",
+      "Mathlib" / "RingTheory",
+      "Mathlib" / "GroupTheory",
+      "Mathlib" / "RepresentationTheory",
+      "Mathlib" / "LinearAlgebra"] },
+  { label := "t-algebraic-geometry",
+    dirs := #[
+      "Mathlib" / "AlgebraicGeometry",
+      "Mathlib" / "Geometry" / "RingedSpace"] },
+  { label := "t-analysis" },
+  { label := "t-category-theory" },
+  { label := "t-combinatorics" },
+  { label := "t-computability" },
+  { label := "t-condensed" },
+  { label := "t-data" },
+  { label := "t-differential-geometry",
+    dirs := #["Mathlib" / "Geometry" / "Manifold"] },
+  { label := "t-dynamics" },
+  { label := "t-euclidean-geometry",
+    dirs := #["Mathlib" / "Geometry" / "Euclidean"] },
+  { label := "t-linter",
+    dirs := #["Mathlib" / "Tactic" / "Linter"] },
+  { label := "t-logic",
+    dirs := #[
+      "Mathlib" / "Logic",
+      "Mathlib" / "ModelTheory"] },
+  { label := "t-measure-probability",
+    dirs := #[
+      "Mathlib" / "MeasureTheory",
+      "Mathlib" / "Probability",
+      "Mathlib" / "InformationTheory"] },
+  { label := "t-meta",
+    dirs := #[
+      "Mathlib" / "Control",
+      "Mathlib" / "Lean",
+      "Mathlib" / "Mathport",
+      "Mathlib" / "Tactic",
+      "Mathlib" / "Util"],
+    exclusions := #["Mathlib" / "Tactic" / "Linter"] },
+  { label := "t-number-theory" },
+  { label := "t-order" },
+  { label := "t-set-theory" },
+  { label := "t-topology",
+    dirs := #[
+      "Mathlib" / "Topology",
+      "Mathlib" / "AlgebraicTopology"] },
+  { label := "CI",
+    dirs := #[".github" / "workflows"] },
```

Comment:
```suggestion
    dirs := #[".github"] },
```
the "main build" file is now in `.github` (although modifying it implies also modifying files inside `workflows`).

We catch a couple of other files in `.github` but not in `.github/workflows` as well.